src/main/java/com/adroll/cantor/HLLCounter.java

```diff
@@ -311,11 +311,21 @@ public void combine(HLLCounter h) {
     a = getAlpha(m);
 
     if(intersectable && h.isIntersectable()) {
-      ts.addAll(h.getMinHash());
-      int curSize = ts.size();
       int mink = Math.min(k, h.getK());
-      for(int i = 0; i < (curSize - mink); i++) {
-        ts.pollLast();
+      Iterator<Long> it = h.getMinHash().iterator();
+      if (ts.size() + h.getMinHash().size() < mink) {
+        ts.addAll(h.getMinHash());
+      } else {
```

Comment:
@mattusifer correctly pointed out a bug in this PR. Fixing the bug revealed that there is no performance gain from this method. Closing